src/libsync/owncloudpropagator.cpp

```diff
@@ -1112,4 +1130,43 @@ QString OwncloudPropagator::remotePath() const
 {
     return _remoteFolder;
 }
+
+PropagateUpdateMetaDataJob::PropagateUpdateMetaDataJob(OwncloudPropagator *propagator, const SyncFileItemPtr &item)
+    : PropagateItemJob(propagator, item)
+{
+}
+
+void OCC::PropagateUpdateMetaDataJob::start()
+{
+    // Update the database:  New remote fileid or Etag or RemotePerm
+    // Or for files that were detected as "resolved conflict".
+    // Or a local inode/mtime change
+
+    // In case of "resolved conflict": there should have been a conflict because they
+    // both were new, or both had their local mtime or remote etag modified, but the
+    // size and mtime is the same on the server.  This typically happens when the
+    // database is removed. Nothing will be done for those files, but we still need
+    // to update the database.
+
+    const QString filePath = propagator()->fullLocalPath(_item->destination());
+    if (_item->_direction == SyncFileItem::Down) {
+        // If the 'W' remote permission changed, update the local filesystem
+        SyncJournalFileRecord prev;
+        if (propagator()->_journal->getFileRecord(_item->_file, &prev)
+            && prev.isValid()
+            && prev._remotePerm.hasPermission(RemotePermissions::CanWrite) != _item->_remotePerm.hasPermission(RemotePermissions::CanWrite)) {
+            const bool isReadOnly = !_item->_remotePerm.isNull() && !_item->_remotePerm.hasPermission(RemotePermissions::CanWrite);
+            FileSystem::setFileReadOnlyWeak(filePath, isReadOnly);
+            if (_item->_checksumHeader.isEmpty()) {
+                _item->_checksumHeader = prev._checksumHeader;
+            }
+            _item->_serverHasIgnoredFiles |= prev._serverHasIgnoredFiles;
+        }
+    }
+    const auto result = propagator()->updateMetadata(*_item);
+    if (!result) {
+        done(SyncFileItem::SoftError, tr("Could not update file : %1").arg(result.error()));
```

Comment:
In this case, the file is possibly treated by `setFileReadOnlyWeak` already, but the subsequent metadata update failed. Is it neccessary here to set the read-only flag back to be conform with the previous database entry?

Reply:
Hmm I assume having file permissions that differ from the db don't really hurt that much.
Also the step should be retried so once it succeeded we should be back in sync.